### DISCUSSION TITLE
How to mock axios put or post request with onUploadProgress config using Mock Service Worker

### DISCUSSION THREAD

**Discussion by chetankondawle** [axios/axios#5705] [Q&A]
I have an file upload API, which has a config to check the progress of file being uplaoded using onUploadProgress. It takes a callback, which takes a progressEvent as ain argument, which gives me total file size and uploaded file size.

I tried mocking this particular upload api request with MSW. MSW does not intercept the request and directly the catch block is called in test case. When, I remvoe the onUploadProgress config, MSW intercepts the request and returns the resposne correctly.

How to mock the request which has onUploadProgress config using Mock Service Worker?

Here is the code snippet for axios config.
```
axiosClient.put(
    `/uploadFile/files`,
    formData,
    {
        timeout: 35000,
        headers: {
            "Content-Type": "multipart/form-data",
        },
        onUploadProgress: (progressEvent) => {
            const progress: number = Math.round(
                (progressEvent.loaded * 100) / progressEvent.total
            );
            console.log("progress");
            setProgressMap((prev) => new Map(prev.set(fileName, progress)));
        },
    }).then((res) => {
        console.log("Here");
        return res.data;
    }).catch((error) => {
        console.log("error);
    });
```

Form data holds the file object .

**Comment by kettanaito** [comment]
Hi, @chetankondawle. Have you tried to reproduce this on `msw@next`? We've made a significant improvement in that version, which is going to become MSW 2.0 sometime later this year. Some of those improvements concern XHR and upload/progress handling in Node.js. There is a decent chance the issue will be fixed once you update:

```sh
npm i msw@next
```

Please note that it's a breaking change and you can find detailed [Migration guideline](https://github.com/mswjs/msw/blob/feat/standard-api/MIGRATING.md) on GitHub. Let me know how it goes.

**Reply by chetankondawle** [reply]
@kettanaito Thank you for reaching out here and over stackoverflow as well. I'll take a look as soon as i get chance, lot of tests are pending. Will reply once I test it

**Reply by kettanaito** [reply]
I'm positive the issue will be fixed once you migrate to the newer version of MSW. If it's not, please let me know and share a minimal reproduction repo so we could fix it together. Thanks!
